From d7bd238f8386563916e18532ad0af2a8cc37f5e9 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 24 Aug 2026 06:23:46 +0000
Subject: [PATCH 6/8] ARM: dts: am437x-cm-t43: describe SPI NOR partitions

Place the CM-T43 SPI NOR partitions under a fixed-partitions container so the MTD core discovers the U-Boot, environment, and reserved regions using the current binding.

Upstream-Status: Pending
Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm/boot/dts/ti/omap/am437x-cm-t43.dts | 33 ++++++++++++---------
 1 file changed, 19 insertions(+), 14 deletions(-)

diff --git a/arch/arm/boot/dts/ti/omap/am437x-cm-t43.dts b/arch/arm/boot/dts/ti/omap/am437x-cm-t43.dts
index e1f3ce0bdb49..6af14994141f 100644
--- a/arch/arm/boot/dts/ti/omap/am437x-cm-t43.dts
+++ b/arch/arm/boot/dts/ti/omap/am437x-cm-t43.dts
@@ -310,24 +310,29 @@ &spi0 {
 	dma-names = "tx0", "rx0";
 
 	flash: flash@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
 		compatible = "jedec,spi-nor";
 		reg = <0>;
 		spi-max-frequency = <20000000>;
-		partition@0 {
-			label = "uboot";
-			reg = <0x0 0xc0000>;
-		};
-
-		partition@c0000 {
-			label = "uboot environment";
-			reg = <0xc0000 0x40000>;
-		};
 
-		partition@100000 {
-			label = "reserved";
-			reg = <0x100000 0x100000>;
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x0 0xc0000>;
+			};
+
+			partition@c0000 {
+				label = "uboot environment";
+				reg = <0xc0000 0x40000>;
+			};
+
+			partition@100000 {
+				label = "reserved";
+				reg = <0x100000 0x100000>;
+			};
 		};
 	};
 };
-- 
2.43.0
